include/engine/datafacade/internal_datafacade.hpp

```diff
@@ -174,46 +177,42 @@ class InternalDataFacade final : public BaseDataFacade
         util::SimpleLogger().Write() << "Data checksum is " << m_check_sum;
     }
 
-    void LoadNodeAndEdgeInformation(const boost::filesystem::path &nodes_file,
-                                    const boost::filesystem::path &edges_file)
+    void LoadNodeAndEdgeInformation(const boost::filesystem::path &nodes_file_path,
+                                    const boost::filesystem::path &edges_file_path)
     {
-        boost::filesystem::ifstream nodes_input_stream(nodes_file, std::ios::binary);
+        boost::filesystem::ifstream nodes_input_stream(nodes_file_path, std::ios::binary);
+        if (!nodes_input_stream)
+        {
+            throw util::exception("Could not open " + nodes_file_path.string() + " for reading.");
+        }
 
-        extractor::QueryNode current_node;
-        unsigned number_of_coordinates = 0;
-        nodes_input_stream.read((char *)&number_of_coordinates, sizeof(unsigned));
+        std::uint32_t number_of_coordinates = storage::io::readNodesSize(nodes_input_stream);
         m_coordinate_list.resize(number_of_coordinates);
         m_osmnodeid_list.reserve(number_of_coordinates);
-        for (unsigned i = 0; i < number_of_coordinates; ++i)
+        storage::io::readNodesData(
+            nodes_input_stream, m_coordinate_list.data(), m_osmnodeid_list, number_of_coordinates);
+
+        boost::filesystem::ifstream edges_input_stream(edges_file_path, std::ios::binary);
+        if (!edges_input_stream)
         {
-            nodes_input_stream.read((char *)&current_node, sizeof(extractor::QueryNode));
-            m_coordinate_list[i] = util::Coordinate(current_node.lon, current_node.lat);
-            m_osmnodeid_list.push_back(current_node.node_id);
-            BOOST_ASSERT(m_coordinate_list[i].IsValid());
+            throw util::exception("Could not open " + edges_file_path.string() + " for reading.");
         }
-
-        boost::filesystem::ifstream edges_input_stream(edges_file, std::ios::binary);
-        unsigned number_of_edges = 0;
-        edges_input_stream.read((char *)&number_of_edges, sizeof(unsigned));
+        auto number_of_edges = storage::io::readEdgesSize(edges_input_stream);
         m_via_geometry_list.resize(number_of_edges);
         m_name_ID_list.resize(number_of_edges);
         m_turn_instruction_list.resize(number_of_edges);
         m_lane_data_id.resize(number_of_edges);
         m_travel_mode_list.resize(number_of_edges);
         m_entry_class_id_list.resize(number_of_edges);
```

Comment:
just a heads up, I am just about to merge the turn-angle branch which adds to this list.